### ISSUE TITLE
Presto driver - user cancelled queries don't get cancelled on Presto

### ISSUE THREAD

**Issue by harshith-bolar-rapido** [metabase/metabase#47555]
### Describe the bug

When I execute a query using the default presto driver shipped with metabase and cancel the query from the UI, the query keeps running on Presto. Ideally this query should have gotten cancelled on Presto. 

This has resulted in increased resource usage as many user cancelled queries keep executing and using up resources.

### To Reproduce

1. Go to New -> Native query
2. Select Presto driver
3. Execute any long running query
4. Cancel the query in between
5. Check Presto UI, the query keeps running

### Expected behavior

User cancelled queries from the UI should be cancelled on Presto

### Logs

I see these errors in the logs when the query is cancelled from the UI

```
[2e8b8166-a68a-4869-849a-a2eeb401d2db] 2024-09-03T16:26:41+05:30 WARN metabase.driver.sql-jdbc.execute Statement cancelation failed.
[2e8b8166-a68a-4869-849a-a2eeb401d2db] 2024-09-03T16:26:41+05:30 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: Broken pipe
```

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.23+9",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.23",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.23+9",
    "os.name": "Linux",
    "os.version": "6.1.85+",
    "user.language": "en",
    "user.timezone": "Asia/Calcutta"
  },
  "metabase-info": {
    "databases": [
      "h2",
      "bigquery-cloud-sdk",
      "presto-jdbc",
      "mysql",
      "postgres",
      "starburst"
    ],
    "hosting-env": "unknown",
    "application-database": "mysql",
    "application-database-details": {
      "database": {
        "name": "MySQL",
        "version": "8.0.31-google"
      },
      "jdbc-driver": {
        "name": "MariaDB Connector/J",
        "version": "2.7.10"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2024-06-13",
      "tag": "v0.50.5-SNAPSHOT",
      "hash": "15343a3"
    },
    "settings": {
      "report-timezone": "Asia/Kolkata"
    }
  }
}
```

### Severity

Severe - As this increases resource usage on Presto blocking execution of actual queries

### Additional context

This started happening after a recent upgrade from 0.39.6 to 0.50

**Comment by paoliniluis** [comment]
related to https://github.com/metabase/metabase/issues/45571

**Comment by sabyasachinandy** [comment]
Thanks for acting on this ! We will check this out

**Comment by iethree** [comment]
🚀 This should also be released by [v0.51](https://github.com/metabase/metabase/milestone/231)
